Sample repository ontology

zhongh opened the pull request.

zhongh committed:
Creating new ttl file for dco-sample ontology

zhongh committed:
Adding class  dco:Repository

Commit:
Repository added to dco ontology

mrpatrickwest commented on dco.ttl:
```diff
@@ -790,84 +790,66 @@ dco:vesselName
               [ a       owl:Restriction ;
-                owl:allValuesFrom rdfs:Literal ;
-                owl:onProperty dco:hasMaterial
+                owl:allValuesFrom core:GeographicLocation ;
```
This doesn't seem quite right. Might also be vivo:Address or other geographic representations. Just not sure how to represent it here.

Commit:
Beginning of dco sample ontology

mrpatrickwest commented:
Example sample: https://udco.tw.rpi.edu/vivo/display/n11018
Example repository: https://udco.tw.rpi.edu/vivo/display/n11180

Commit:
Added sample groups and validated

Added SampleGroup class and related properties. Also made sure that all of the .ttl files were valid

Commit:
Ontology after discussions Apr 04, 2016

Commit:
updated properties and classes match cmap

Added the sampling ontology from csiro in sam.ttl and sam-vivo.ttl. Made sure all of the properties in dco have corresponding entries in dco-vivo.ttl. Added
isCitedBy in dco ontology and restriction in dco-sample

mrpatrickwest commented:
Problem with pulling in the sampling ontology is that it imports quite a few different ontologies

https://udco.tw.rpi.edu/vivo/listOntologies

All the ones on this page that do not have a namespace label.

Should we add labels and annotations so they show up looking good?

Commit:
Small changes to sample ontology, fixed ttl file

Added property group for a property and fixed the ttl file for dco-sample.ttl, adding namespace.

Commit:
Misc vivo properties on non-dco non-vivo ont

vivo properties for dc and dcterms ontology and properties. Corrected relatedProject and isCitedBy annotations

mrpatrickwest commented:
https://udco.tw.rpi.edu/vivo/display/n10039 - one of the repositories entered into UDCO ... not that bad. Please review.

mrpatrickwest commented:
@zednis @olyerickson @zhonghao1014 @am-e @xgmachina 

please review

Commit:
Switched from full sampling ontology to samfl

sampling feature lite

Commit:
Changed URI of sam lite ontology in other files

Was still using the full sampling feature ontology uri, fixed that. Also missed the xsd prefix in dco-misc.

mrpatrickwest commented:
New example sample: https://udco.tw.rpi.edu/vivo/display/n7813
New example repository: https://udco.tw.rpi.edu/vivo/display/n22533

Commit:
Missing property igsnRegistrant

Off of IGSNRegistration

zednis commented on dco-sample.ttl:
```diff
@@ -0,0 +1,184 @@
+dcosample:SampleGroup
+      a       owl:Class ;
+      rdfs:label "Sample Group"@en-US ;
+      rdfs:subClassOf dco:Object ;
```
Should this also be a subclass of [`samfl:SamplingCollection`?](http://def.seegrid.csiro.au/ontology/om/sam-lite#SamplingCollection)

mrpatrickwest replied:
yes it should be

zednis commented on dco-sample.ttl:
```diff
@@ -0,0 +1,184 @@
+          A sample classification or annotation
+      """^^rdf:XMLLiteral .
+
+dcosample:hasIGSNRegistration
```
Should there be an inverse property for this property?

The IGSN Registration page has no link to the registered sample/feature https://udco.tw.rpi.edu/vivo/display/n16187

mrpatrickwest replied:
yes there should be an inverse property.

Commit:
Response to github comments

Changed the namespace label of the sam-lite ontology to samfl.
Change the label of the sam-lite ontology to Sampling Feature.
Added inverse property of hasIGSNRegistration called registrationForSample and corresponding vivo properties

Commit:
Added inverse of includesSampleGroup

To link the sample group back to the repository. Also allowing new instances of sample group to be created from a repository instead of having to select
from existing ones.

Commit:
igsn registration on sampling feature

since an igsn can be assigned to any sampling feature changed the domain from Specimen to SamplingFeature. Also changed the inverse property to
registrationForSamplingFeature.

xgmachina commented:
How about mapping a few properties to be sub-property of dct:hasPart and others to dct:isPartOf

mrpatrickwest commented:
YOu mean instead of using hasPart and isPartOf directly?

zednis commented:
Are we using hasPart and isPartOf directly?  I wasn't sure with `dcosample:includesSample` in the ontology.

`dcterms:isPartOf` instead of `dcosample:includesSample`?  Or make `dcosample:includesSample` a sub-property of `dcterms:isPartOf`?

mrpatrickwest commented:
Not sure why I did it this way, but SampleGroup -> hasPart -> Sample -> isPartOf -> SampleGroup

mrpatrickwest commented:
seems that should just be includesSample, remove the domain, add as restriction to SampleRepository and SampleGroup

mrpatrickwest commented:
and make includesSample and includesSampleGroup subproperties of hasPart, and the inverse a subproperty of isPartOf.

zednis commented:
> Not sure why I did it this way, but SampleGroup -> hasPart -> Sample -> isPartOf -> SampleGroup

This makes sense, though if SampleGroup is a `samfl:SamplingCollection` then it should use `samfl:member`.

```
samfl:SamplingCollection
  rdf:type owl:Class ;
  dct:description "Collection of features on which observations may be made, which are intended to be representative of one or more features that are not fully accessible for some reason. "^^xsd:string ;
  rdfs:comment "The member sampling features are intended to sample domain features, as indicated by one or more sampledFeature properties. If only one is present then this is the sampled feature of all the member sampling features, unless locally overridden."^^xsd:string ;
  rdfs:isDefinedBy <http://def.seegrid.csiro.au/ontology/om/sam-lite> ;
  rdfs:label "Collection of sampling features"@en ;
  rdfs:subClassOf [
      rdf:type owl:Restriction ;
      owl:minCardinality "1"^^xsd:nonNegativeInteger ;
      owl:onProperty samfl:member ;
    ] ;
.
```

zednis commented:
> and make includesSample and includesSampleGroup subproperties of hasPart, and the inverse a subproperty of isPartOf.

:+1:

mrpatrickwest commented:
okay ... to summarize

includesSample -> sub property of -> hasPart (inerse sub property of isPartOf)
includesSampleGroup -> sub property of -> hasPart (inerse sub property of isPartOf)

SampleGroup remove restrictions for hasPart since there's already a property samfl:member

Did I get that right?

everything is good.

mrpatrickwest commented:
no inverse property for samfl:member so would add dcosample:memberOf to be the inverse property

zhonghao1014 commented:
👍

zednis commented:
> includesSample -> sub property of -> hasPart (inerse sub property of isPartOf)
> includesSampleGroup -> sub property of -> hasPart (inerse sub property of isPartOf)

I think this is good; just to sanity check - this is the definition for `dcterms:hasPart`

> A related resource that is included either physically or logically in the described resource.

Do we all agree this makes sense as a super-property of includesSample and includesSampleGroup?
- does it make sense to say a sample is included physically or logically in the repository?
- does it make sense to say a sample group is included physically or logically in the repository?

mrpatrickwest commented:
yes, I think that makes sense.

xgmachina commented:
yes, too on the use of dct:hasPart

Commit:
hasPart and isPartOf sub properties

includesSample and includesSampleGroup are subproperties of hasPart.
Their inverses are sub properties of isPartOf

Commit:
Added sampleCurationPractice

am-e commented:
I like the use of dct:hasPart and dct:isPartOf, does that take care of our previous discussion about the relationship between samples/sample groups and the repository?

Commit:
Added comments to properties that did not have one
